llama_index/indices/document_summary/retrievers.py

```diff
@@ -0,0 +1,213 @@
+"""Document summary retrievers.
+
+This module contains retrievers for document summary indices.
+
+"""
+
+from llama_index.callbacks.schema import CBEventType
+from llama_index.indices.document_summary.base import GPTDocumentSummaryIndex
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.indices.base_retriever import BaseRetriever
+from typing import Any, List, Optional, Callable, Tuple, Dict
+from llama_index.data_structs.node import Node, NodeWithScore
+from llama_index.prompts.prompts import QuestionAnswerPrompt
+from llama_index.indices.service_context import ServiceContext
+from llama_index.indices.query.embedding_utils import (
+    get_top_k_embeddings,
+)
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_CHOICE_SELECT_PROMPT_TMPL = (
+    "A list of documents is shown below. Each document has a number next to it along "
+    "with a summary of the document. A question is also provided. \n"
+    "Select the numbers of the documents "
+    "you should consult to answer the question, in order of relevance. \n"
+    "Do not include any documents that are not relevant to the question. \n"
+    "Example format: \n"
+    "Document 1: <summary of document 1>\n"
+    "Document 2: <summary of document 2>\n"
+    "..."
+    "Document 10: <summary of document 10>\n"
+    "Question: <question>\n"
+    "Answer: 9, 7, 4\n\n"
```

Comment:
Another alternative to consider: 
```
Respond with how relevant on a scale from 1-10 you think that document is, only giving a document if it's relevance is greater than {{minimum_relevance}}.
```

Might complicate the interface, but giving a relevance score allows things like choosing a minimum relevance for recall vs precision trade-offs, or adding something like `similarity_top_k` to choose the most similar K.